airflow/models/dag.py

```diff
@@ -2730,22 +2730,50 @@ def add_logger_if_needed(ti: TaskInstance):
         )
 
         tasks = self.task_dict
+        # Special case when retry is set and DAG is paused
+        task_try_number = {}
+
         self.log.debug("starting dagrun")
         # Instead of starting a scheduler, we run the minimal loop possible to check
         # for task readiness and dependency management. This is notably faster
         # than creating a BackfillJob and allows us to surface logs to the user
         while dr.state == State.RUNNING:
             schedulable_tis, _ = dr.update_state(session=session)
-            try:
-                for ti in schedulable_tis:
+
+            for ti in schedulable_tis:
+                try:
                     add_logger_if_needed(ti)
                     ti.task = tasks[ti.task_id]
-                    _run_task(ti, session=session)
-            except Exception:
-                self.log.info(
-                    "Task failed. DAG will continue to run until finished and be marked as failed.",
-                    exc_info=True,
-                )
+                    if ti.task_id not in task_try_number:
+                        task_try_number[ti.task_id] = 0
```

Comment:
Should we make this dict consider map_index instead? If there are multiple map indexes (dynamic task mapping), they would have the same task_id and overwrite each other in this dict.

Reply:
Also why is this `if` block needed? You use `get()` below so the task_id shouldn’t need to be in the dict at this point anyway.